File: exercise01/part1.py
import unittest


def produce_sum(seq):
    if len(seq) == 0:
        return 0

    previous = seq[-1]
    sum = 0
    for i in range(0, len(seq)):
        if seq[i] == previous:
            sum += ord(seq[i]) - ord('0')
        previous = seq[i]

    return sum


class TestProduceSum(unittest.TestCase):
    def test_produce_sum(self):
        self.assertEqual(produce_sum('1122'), 3)
        self.assertEqual(produce_sum('1111'), 4)
        self.assertEqual(produce_sum('1234'), 0)
        self.assertEqual(produce_sum('91212129'), 9)


if __name__ == '__main__':
    with open('input.txt', 'r') as f:
        data = f.read()
        print produce_sum(data)
    unittest.main()

File: exercise01/part1.cc
#include <fstream>
#include <iostream>

using namespace std;

int produce_sum(string seq)
{
    if (seq.length() == 0)
        return 0;
    int sum = 0;
    char previous = seq[seq.length()-1];
    for (int i=0; i <seq.length(); ++i)
    {
        char a = seq[i];
        if (a == previous)
            sum += a - '0';
        previous = a;
    }
    return sum;
}


int main() 
{
    std::ifstream infile("input.txt");
    string line;
    std::getline(infile, line);
    cout << produce_sum(line) << endl;
    cout << produce_sum("1122") << "==" << 3 << endl;
    cout << produce_sum("1111") << "=="  << 4 << endl;
    cout << produce_sum("1234") << "=="  << 0 << endl;
    cout << produce_sum("91212129") << "=="  << 9 << endl;
    return 0;
}
